### PR TITLE
Automated release pipeline

### PR DESCRIPTION
**Pull request by leordev**
- Add a release workflow.
- Release instructions added on the README file.

Working Evidence: https://github.com/TBD54566975/web5-swift/releases/tag/0.0.3-test (let's remove it before merging it!)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    # validate if input was versionType = Major, Minor, Patch or Manual
+    # if its Manual we require the `customVersion` value
+    if: github.event.inputs.versionType != 'manual' || github.event.inputs.customVersion != null
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
```
Comment (amika-sq) [inline .github/workflows/release.yml:31]: Curious if we can/should reduce this. Would we be able to get away with 1 here?

**Review by amika-sq — APPROVED** [review]
Looks great! Just one minor comment, not blocking at all though.

File: .github/workflows/release.yml (hunk 1 above)
Comment (leordev) [reply .github/workflows/release.yml:31]: I did a little research and I think the best we can get away with is by using `filter: tree:0`:
https://github.com/actions/checkout/issues/1152#issuecomment-1752251377

The reason is that we need the history to be able to figure out the latest git tag. A shallow clone unfortunately does not give us that.

Maybe we can reduce the depth to 50, or 25 assuming that we will always cut new releases when we are about 50 or 25 merged PRs. WDYT?

Comment (amika-sq) [reply .github/workflows/release.yml:31]: Gotcha! I think it's fine to leave at 100 then, as that should usually contain the last tag. Probably a bit better to overestimate than under :)

**Pull request merged**